src/core/FileUpload/FileUpload.tsx

```diff
@@ -0,0 +1,115 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { useTheme } from '../utils/hooks/useTheme';
+import { CommonProps } from '../utils/props';
+import { useMergedRefs } from '../utils/hooks/useMergedRefs';
+import '@itwin/itwinui-css/css/file-upload.css';
+
+export type FileUploadProps = {
+  /**
+   * Content shown over `children` when file is being dragged onto the component.
+   */
+  content: React.ReactNode;
+  /**
+   * Callback fired when files are dropped onto the component.
+   */
+  onFileDropped: (files: FileList) => void;
```

Comment:
Cant it be optional? Or you wanted to enforce it, cause it doesnt make sense not to listen to the event?

Reply:
Yeah I wanted to enforce it because there's no point to this component without this callback

Reply:
hmm.. I just saw in your tests, that there was a point (only for visual check, ofc, but still :D)